fabric-registry-sync-v0/src/main/java/net/fabricmc/fabric/mixin/registry/sync/MixinPlayerManager.java

```diff
@@ -17,27 +17,37 @@
 package net.fabricmc.fabric.mixin.registry.sync;
 
 import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Unique;
 import org.spongepowered.asm.mixin.injection.At;
 import org.spongepowered.asm.mixin.injection.Inject;
 import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
 
 import net.minecraft.network.ClientConnection;
-import net.minecraft.network.Packet;
+import net.minecraft.network.PacketByteBuf;
 import net.minecraft.server.PlayerManager;
 import net.minecraft.server.network.ServerPlayerEntity;
 
+import net.fabricmc.fabric.api.network.ServerSidePacketRegistry;
 import net.fabricmc.fabric.impl.registry.sync.RegistrySyncManager;
 
 @Mixin(PlayerManager.class)
 public abstract class MixinPlayerManager {
+	@Unique
+	private volatile PacketByteBuf currentSyncData;
+
 	@Inject(method = "onPlayerConnect", at = @At(value = "INVOKE", target = "Lnet/minecraft/network/packet/s2c/play/DifficultyS2CPacket;<init>(Lnet/minecraft/world/Difficulty;Z)V"))
-	public void onPlayerConnect(ClientConnection lvt1, ServerPlayerEntity lvt2, CallbackInfo info) {
+	public void onPlayerConnect(ClientConnection connection, ServerPlayerEntity player, CallbackInfo info) {
 		// TODO: If integrated and local, don't send the packet (it's ignored)
 		// TODO: Refactor out into network + move registry hook to event
-		Packet<?> packet = RegistrySyncManager.createPacket();
+		PacketByteBuf data = this.currentSyncData;
+
+		// Cover registry changes
+		if (data == null) {
+			this.currentSyncData = data = RegistrySyncManager.createData();
+		}
 
-		if (packet != null) {
-			lvt2.networkHandler.sendPacket(packet);
+		if (data != null) {
+			player.networkHandler.sendPacket(ServerSidePacketRegistry.INSTANCE.toPacket(RegistrySyncManager.ID, new PacketByteBuf(data.duplicate())));
```

Comment:
you can just send the packet with the server side packet registry really, assuming the server play network handler has already been created